jf_agent/git/__init__.py

```diff
@@ -261,28 +264,56 @@ def get_git_client(config, creds):
 
 @diagnostics.capture_timing()
 @agent_logging.log_entry_exit(logger)
-def load_and_dump_git(config, endpoint_git_instance_info, git_connection):
+def load_and_dump_git(
+    config: GitConfig,
+    endpoint_git_instance_info: dict,
+    outdir: str,
+    compress_output_files: bool,
+    git_connection,
+):
+
+    # use the unique git instance agent key to collate files
+    instance_slug = endpoint_git_instance_info['slug']
+    instance_key = endpoint_git_instance_info['key']
+    outdir = f'{outdir}/git_{instance_key}'
+    os.mkdir(outdir)
```

Comment:
Wondering about the rationale for using subdirectories for each instance. Are you sure it's necessary/desirable? If we guarantee that the key is unique, should we just write all files into a single flat directory instead?